From 2e887e76dd192a642544746caf431a652e98fbe9 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Wed, 27 Feb 2019 10:15:43 -0800
Subject: [PATCH 0519/1851] dt-bindings: display: xlnx: Add clock framework
 support to VPSS Scaler bridge driver

commit 3ccda000eba86036f7716235a7bf7295cd17a9e3 from
https://github.com/Xilinx/linux-xlnx.git

This patch adds support for clock framework to VPSS Scaler bridge driver
by adding and making clocks and clock-names as required properties.

This breaks backwards compatibility as the IP version hasn't changed.
So compatible string isn't changed. But new required clock properties
are added.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../devicetree/bindings/display/xlnx/xlnx,vpss-scaler.txt  | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Documentation/devicetree/bindings/display/xlnx/xlnx,vpss-scaler.txt b/Documentation/devicetree/bindings/display/xlnx/xlnx,vpss-scaler.txt
index f3e30e7daa75..312c82d7ee25 100644
--- a/Documentation/devicetree/bindings/display/xlnx/xlnx,vpss-scaler.txt
+++ b/Documentation/devicetree/bindings/display/xlnx/xlnx,vpss-scaler.txt
@@ -24,6 +24,11 @@ Required properties:
 
 - reset-gpios: GPIO specifier to assert/de-assert the reset line.
 
+- clocks: List of phandles to AXI Lite and Video clock
+
+- clock-names: Must contain "aclk_ctrl" and "aclk_axis" in same order as clocks
+  listed in clocks property.
+
 Example:
 	scaler@a0040000 {
 		compatible = "xlnx,vpss-scaler";
@@ -32,4 +37,6 @@ Example:
 		xlnx,num-hori-taps = <8>;
 		xlnx,num-vert-taps = <8>;
 		xlnx,pix-per-clk = <2>;
+		clock-names = "aclk_ctrl", "aclk_axis";
+		clocks = <&misc_clk_0>, <&misc_clk_1>;
 	}
-- 
2.31.1

